cplusplus/VImage.cpp

```diff
@@ -427,73 +427,69 @@ set_property( VipsObject *object, const char *name, const GValue *value )
 void
 VOption::set_operation( VipsOperation *operation )
 {
-	std::list<Pair *>::iterator i;
-
-	for( i = options.begin(); i != options.end(); ++i )
-		if( (*i)->input ) {
+	for( auto&& option : options )
```

Comment:
auto and this form of fopr loop are C++11 features, which many older compilers do not support. At some point we should switch to requiring a C++11 compiler, but I don't know if we're at that point yet. 

Perhaps we should have a discussion about transitioning to C++11?